sdk/core/Azure.Core/samples/Events.md

```diff
@@ -0,0 +1,190 @@
+# Azure.Core Event samples
+
+**NOTE:** Samples in this file only apply to packages following the
+[Azure SDK Design Guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html).
+The names of these packages usually start with `Azure`.
+
+Most Azure client libraries offer both synchronous and asynchronous methods for
+calling Azure services.  You can distinguish the asynchronous methods by their
+`Async` suffix.  For example, `BlobClient.Download` and `BlobClient.DownloadAsync`
+make the same underlying REST call and only differ in whether they block.  We
+recommend using our async methods for new applications, but there are perfectly
+valid cases for using sync methods as well.  These dual method invocation
+semantics work great, but require a little extra care when writing event handlers.
+
+The `SyncAsyncEventHandler` is a delegate used by events in Azure client
+libraries to represent an event handler that can be invoked from either sync or
+async code paths.  It takes event arguments deriving from `SyncAsyncEventArgs`
+that contain important information for writing your event handler.
+
+- `SyncAsyncEventArgs.CancellationToken` is a cancellation token related to the
+  original operation that raised the event.  It's important for your handler to
+  pass this token along to any asynchronous or long-running synchronous
+  operations that take a token so cancellation (via something like
+  `new CancellationTokenSource(TimeSpan.FromSeconds(10)).Token`, for example)
+  will correctly propagate.
+
+- There is a `SyncAsyncEventArgs.RunSynchronously` flag indicating whether your
+  handler was invoked synchronously or asynchronously.  In general,
+
+    - If you're calling sync methods on your client, you should use sync methods
+      to implement your event handler.  You can return `Task.CompletedTask`.
+    - If you're calling async methods on your client, you should use async
+      methods where possible to implement your event handler.
+    - If you're not in control of how the client will be used or want to write
+      safer code, you should check the `RunSynchronously` property and call
+      either sync or async methods as directed.
+
+  There are code examples of all three situations below to compare.  Please also
+  see the note at the very end discussing the dangers of sync-over-async to
+  understand the risks of not using the `RunSynchronously` flag.
+
+- Most events will customize the event data by deriving from `SyncAsyncEventArgs`
+  and including details about what triggered the event or providing options to
+  react.  Many times this will include a reference to the client that raised the
+  event in case you need it for additional processing.
+
+When an event using `SyncAsyncEventHandler` is raised, the handlers will be
+executed sequentially to avoid introducing any unintended parallelism.  The
```

Comment:
```suggestion
executed sequentially to avoid introducing any unintended parallelism to calling code.  The
```

Reply:
I think "calling code" muddies it a bit since we're avoiding unintentional parallelism in the handlers.